clients/drcachesim/tracer/syscall_pt_trace.cpp

```diff
@@ -230,6 +233,8 @@ syscall_pt_trace_t::trace_data_dump(drpttracer_output_autoclean_t &output)
         return false;
     }
 
+    uint64_t total_size = PT_DATA_PDB_HEADER_SIZE + data->pt_size;
+    auto to_write = std::unique_ptr<uint8_t[]>(new uint8_t[total_size]);
```

Comment:
It is not safe to call new for static app link setups.  Use placement new instead via DR's allocator.

There is an automated check that should have failed with this -- ah is it because GA never tests BUILD_PT_TRACER?  And your local runs only ran these couple of tests?

This is not a good situation: we need our tests to catch things like this.  I don't remember exactly which tests would catch this: please see  #DR_DISALLOW_UNSAFE_STATIC and dr_disallow_unsafe_static_behavior().  Please try to set it up so that GA will fail on this -- before submitting this PR ideally.

Reply:
BUILD_PT_TRACER is always ON for LINUX AND X86 AND X64. If there's no PT support, it'll just fail if the user tries to trace system calls. So the tests should've failed and this is a regression?

Reply:
The check seems to be performed in non-static builds. I was able to get the existing kernel test to fail by adding `-raw_compress none` (thanks for the suggestion in our offline discussion). I modified the existing test to use that setting.